src/apps/answers/tests/test_answers.py

```diff
@@ -3221,6 +3221,113 @@ async def test_validate_multiinformant_assessment_fail_target_subject_not_found(
         assert response.json()["result"]["valid"] is False
         assert response.json()["result"]["code"] == "invalid_target_subject"
 
+    async def test_validate_multiinformant_assessment_fail_temporary_relation_expired(
+        self, client, tom: User, applet_one: AppletFull, session: AsyncSession, tom_applet_one_subject
+    ):
+        client.login(tom)
+
+        subject_service = SubjectsService(session, tom.id)
+
+        source_subject = await subject_service.create(
+            SubjectCreate(
+                applet_id=applet_one.id,
+                creator_id=tom.id,
+                first_name="source",
+                last_name="subject",
+                secret_user_id=f"{uuid.uuid4()}",
+            )
+        )
+        target_subject = await subject_service.create(
+            SubjectCreate(
+                applet_id=applet_one.id,
+                creator_id=tom.id,
+                first_name="target",
+                last_name="subject",
+                secret_user_id=f"{uuid.uuid4()}",
+            )
+        )
+
+        # create a relation between respondent and source
+        await subject_service.create_relation(
+            relation="take-now",
+            source_subject_id=tom_applet_one_subject.id,
+            subject_id=source_subject.id,
+            meta={
+                "expiresAt": (datetime.datetime.now() - datetime.timedelta(days=1)).isoformat(),
+            },
+        )
+        # create a relation between respondent and target
+        await subject_service.create_relation(
+            relation="take-now",
+            source_subject_id=tom_applet_one_subject.id,
+            subject_id=target_subject.id,
+            meta={
+                "expiresAt": (datetime.datetime.now() - datetime.timedelta(days=1)).isoformat(),
+            },
+        )
```

Comment:
`tom` is the owner of `applet_one` so these relations won't matter for the answer validation. Try creating a relation between two non team-member users, like this one: https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/04a411c1e533ada23cd01c4acc5cf63976f6e95d/src/apps/subjects/tests/tests.py#L595-L599

This made me realise that this condition needs to be adjusted https://github.com/ChildMindInstitute/mindlogger-backend-refactor/blob/74e1d2d775f23150569e1376c1b97c9766ee63fc/src/apps/answers/service.py#L389-L396

To this

```python
        if not is_admin:
            if not target_subject or not source_subject:
                raise MultiinformantAssessmentNoAccessApplet("Missing target subject or source subject")

            await self._validate_user_role_for_take_now(applet_id, respondent_subject)
            await self._validate_relation_between_subjects_in_applet(respondent_subject, target_subject, source_subject)
```

Reply:
adding these checks in this way make lots of tests fail because the target_subject or the source_subject are used but they are no initialized which is why I added it this way